docs/pages/deployment/tls-configuration.rst

```diff
@@ -145,58 +145,60 @@ For `NGINX <https://www.nginx.com/>`_ the proxy configuration could look as foll
-      }
+    events {
+        worker_connections  1024;
     }
 
-    server {
-      server_name nuts-n2n;
-      listen                    443 ssl http;
-      ssl_certificate           /etc/nginx/ssl/server.pem;
-      ssl_certificate_key       /etc/nginx/ssl/key.pem;
-      ssl_client_certificate    /etc/nginx/ssl/truststore.pem;
-      ssl_verify_client         on;
-      ssl_verify_depth          1;
-
-      location / {
-        proxy_pass http://nuts-node-http-n2n
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; // for correct IP logging
-      }
+    http {
+        server {
+          server_name nuts-grpc;
+          listen                    5555 ssl http2;
+          ssl_certificate           /etc/nginx/ssl/server.pem;
+          ssl_certificate_key       /etc/nginx/ssl/key.pem;
+          ssl_client_certificate    /etc/nginx/ssl/truststore.pem;
+          ssl_verify_client         on;
+          ssl_verify_depth          1;
+
+          location / {
+            grpc_pass grpc://nuts-node:5555;
+            grpc_set_header X-SSL-CERT $ssl_client_escaped_cert;        # add peer's SSL cert
+            grpc_set_header X-Forwarded-For $proxy_add_x_forwarded_for; # for correct IP logging
+          }
+        }
+
+
+        server {
+          server_name nuts-n2n;
+          listen                    443 ssl;
+          ssl_certificate           /etc/nginx/ssl/server.pem;
+          ssl_certificate_key       /etc/nginx/ssl/key.pem;
+          ssl_client_certificate    /etc/nginx/ssl/truststore.pem;
+          ssl_verify_client         on;
+          ssl_verify_depth          1;
+
+          location /n2n {
+            proxy_pass http://nuts-node:8080;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; # for correct IP logging
+          }
+        }
+
+        server {
+          server_name nuts-public;
+          listen                    443 ssl;
+          ssl_certificate           /etc/nginx/ssl/server.pem;
+          ssl_certificate_key       /etc/nginx/ssl/key.pem;
+
+          location /public {
+            proxy_pass http://nuts-node:80;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; # for correct IP logging
+          }
+        }
     }
 
-    server {
-      server_name nuts-public;
-      listen                    443 ssl http;
-      ssl_certificate           /etc/nginx/ssl/server.pem;
-      ssl_certificate_key       /etc/nginx/ssl/key.pem;
+.. note::
 
-      location / {
-        proxy_pass http://nuts-node-http-public
-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; // for correct IP logging
-      }
-    }
+    During synchronization of a new Nuts node it is possible that the gRPC stream contains messages larger than NGINX is willing to accept.
+    To limit these issues add ``proxy_buffering off`` to the grpc config section, and increase the buffer size to any reasonably large number by adding ``grpc_buffer_size 128M``.
```

Comment:
I think it has to be much, much larger than 128mb? At least from what @woutslakhorst read.